pkg/clusterapi/aws/actuator.go

```diff
@@ -344,44 +344,50 @@ func (a *Actuator) DeleteMachine(machine *clusterv1.Machine) error {
 	return TerminateInstances(client, instances, mLog)
 }
 
-// Update attempts to sync machine state with an existing instance. Today this just updates status
+// Update updates a machine.
+func (a *Actuator) Update(cluster *clusterv1.Cluster, machine *clusterv1.Machine) error {
+	_, err := a.UpdateMachine(cluster, machine)
+	return err
+}
+
+// UpdateMachine attempts to sync machine state with an existing instance. Today this just updates status
 // for details that may have changed. (IPs and hostnames) We do not currently support making any
 // changes to actual machines in AWS. Instead these will be replaced via MachineDeployments.
-func (a *Actuator) Update(cluster *clusterv1.Cluster, machine *clusterv1.Machine) error {
+func (a *Actuator) UpdateMachine(cluster *clusterv1.Cluster, machine *clusterv1.Machine) (*clusterv1.Machine, error) {
```

Comment:
Are we returning the machine just so that we can inspect the updated machine in the tests? Or does it truly make sense to return the machine, from an API standpoint? If it is the former, then we could get the machine from the fake client action instead in the tests. Arguably, that is the side effect that you want to test anyway.

Reply:
Mostly yes, but it also matched the pattern used for Create and Delete. I'm ok with switching it to get it from the fake client action though, will update.